lib/iris/tests/unit/fileformats/netcdf/test_Saver.py

```diff
@@ -612,5 +612,162 @@ def test_laea_cs(self):
+                                      'single_packed_signed.cdl'))
+
+    def test_single_packed_unsigned(self):
+        """Test saving a single CF-netCDF file with packing into unsigned. """
+        # Read PP input file.
+        file_in = tests.get_data_path(
+            ('PP', 'cf_processing',
+             '000003000000.03.236.000128.1990.12.01.00.00.b.pp'))
+        cube = iris.load_cube(file_in)
+        datatype = 'u1'
+        scale_factor, offset = self._get_scale_factor_add_offset(cube,
+                                                                 datatype)
+        # Write Cube to netCDF file.
+        with self.temp_filename(suffix='.nc') as file_out:
+            iris.save(cube, file_out, packing=datatype)
+            decimal = int(-np.log10(scale_factor))
+            packedcube = iris.load_cube(file_out)
+            # Check that packed cube is accurate to expected precision
+            self.assertArrayAlmostEqual(cube.data, packedcube.data,
+                                        decimal=decimal)
+            # Check the netCDF file against CDL expected output.
+            self.assertCDL(file_out, ('unit', 'fileformats', 'netcdf',
+                                      'save',
+                                      'single_packed_unsigned.cdl'))
+
+    def test_single_packed_manual_scale(self):
+        """Test saving a single CF-netCDF file with packing with scale
+        factor and add_offset set manually."""
+        file_in = tests.get_data_path(
+            ('PP', 'cf_processing',
+             '000003000000.03.236.000128.1990.12.01.00.00.b.pp'))
+        cube = iris.load_cube(file_in)
+        datatype = 'i2'
+        scale_factor, offset = self._get_scale_factor_add_offset(cube,
+                                                                 datatype)
+        packdict = dict(dtype=datatype, scale_factor=scale_factor,
+                        add_offset=offset)
+
+        # Write Cube to netCDF file.
+        with self.temp_filename(suffix='.nc') as file_out:
+            iris.save(cube, file_out, packing=packdict)
+            decimal = int(-np.log10(scale_factor))
+            packedcube = iris.load_cube(file_out)
+            # Check that packed cube is accurate to expected precision
+            # Check the netCDF file against CDL expected output.
+            self.assertCDL(file_out, ('unit', 'fileformats', 'netcdf',
+                                      'save',
+                                      'single_packed_manual.cdl'))
+            self.assertArrayAlmostEqual(cube.data, packedcube.data,
+                                        decimal=decimal)
+
+    def test_multi_packed_single_dtype(self):
+        """Test saving multiple packed cubes with the same pack_dtype."""
+        # Read PP input file.
+        file_in = tests.get_data_path(('PP', 'cf_processing',
+                                       'abcza_pa19591997_daily_29.b.pp'))
+        cubes = iris.load(file_in)
```

Comment:
There was a lot more redundancy than just the setup. I removed the redundancy, but without splitting the class.